src/ActionModal/components/ActionModal.vue

```diff
@@ -694,12 +705,7 @@ export default {
       const feeProperties = {
         gasEstimate: this.gasEstimate,
         gasPrice: {
-          // hack
-          amount:
-            this.networkId.startsWith(`terra`) &&
-            this.selectedBalance.denom !== this.network.stakingDenom
-              ? this.gasPrice * this.amount
-              : this.gasPrice,
+          amount: this.estimatedFee / this.gasEstimate,
```

Comment:
👀

Reply:
fees = gasPrice * gasEstimate (in comos-api)
we want: fees = gasPrice * gasEstimate + amount * tax
ergo: gasPriceDesired * gasEstimate = gasPrice * gasEstimate + amount * tax
ergo: gasPriceDesired = (gasPrice * gasEstimate + amount * tax) / gasEstimate

Reply:
added a comment

Reply:
yes, now I saw it. Thank you for this amazing Math clarification 🙏